internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -325,133 +311,86 @@ func (b *BeatsMonitor) injectLogsInput(cfg map[string]interface{}, componentIDTo
 				},
 			},
 			"processors": []interface{}{
+				// copy original dataset so we can drop the dataset field
 				map[string]interface{}{
-					"add_fields": map[string]interface{}{
-						"target": "data_stream",
-						"fields": map[string]interface{}{
-							"type":      "logs",
-							"dataset":   "elastic_agent",
-							"namespace": monitoringNamespace,
-						},
-					},
-				},
-				map[string]interface{}{
-					"add_fields": map[string]interface{}{
-						"target": "event",
-						"fields": map[string]interface{}{
-							"dataset": "elastic_agent",
-						},
-					},
-				},
-				map[string]interface{}{
-					"add_fields": map[string]interface{}{
-						"target": "elastic_agent",
-						"fields": map[string]interface{}{
-							"id":       b.agentInfo.AgentID(),
-							"version":  b.agentInfo.Version(),
-							"snapshot": b.agentInfo.Snapshot(),
-						},
-					},
-				},
-				map[string]interface{}{
-					"add_fields": map[string]interface{}{
-						"target": "agent",
-						"fields": map[string]interface{}{
-							"id": b.agentInfo.AgentID(),
+					"copy_fields": map[string]interface{}{
+						"fields": []interface{}{
+							map[string]interface{}{
+								"from": "data_stream.dataset",
+								"to":   "data_stream.dataset_original",
+							},
 						},
 					},
 				},
+				// drop the dataset field so following copy_field can copy to it
 				map[string]interface{}{
 					"drop_fields": map[string]interface{}{
 						"fields": []interface{}{
-							"ecs.version", //coming from logger, already added by libbeat
+							"data_stream.dataset",
 						},
-						"ignore_missing": true,
-					},
-				}},
-		},
-	}
-	for unit, binaryName := range componentIDToBinary {
```

Comment:
i'm just wondering, have you checked overall CPU consumption with the move towards stdout/err monitoring? 
seems like std should be a bit better right?

Reply:
I don't know about the overall load change, I would say its a net wash. Being that the components no longer need to worry about rotation, syncing files, or worrying about deleting old logs.

I think the benefits of components not needing to really have to worry about logging, rotation, etc; is a win!